src/board/compass.ts

```diff
@@ -0,0 +1,33 @@
+import { RangeSensor, State } from "./state";
+
+type StateKeys = "compassX" | "compassY" | "compassZ" | "compassHeading";
+
+export class Compass {
+  state: Pick<State, "compassX" | "compassY" | "compassZ" | "compassHeading">;
+
+  constructor() {
+    const min = -2_000_000;
+    const max = 2_000_000;
+    this.state = {
+      compassX: new RangeSensor("compassX", min, max, 0, "nT"),
+      compassY: new RangeSensor("compassY", min, max, 0, "nT"),
+      compassZ: new RangeSensor("compassZ", min, max, 0, "nT"),
+      compassHeading: new RangeSensor("compassHeading", 0, 360, 0, "deg"),
```

Comment:
Unit isn't documented in CODAL that I can see but MakeCode uses uBit.compass.heading() directly as as degrees value so this is all good.